extensions/vertx-http/deployment/src/main/java/io/quarkus/devui/deployment/menu/AuthProcessor.java

```diff
@@ -0,0 +1,274 @@
+                var isPolicy = propertyKey.startsWith("quarkus.http.auth.policy");
+
+                if (!(isPermission || isPolicy))
+                    continue;
+
+                Optional<ConfigDescriptionBuildItem> optionalConfigDescriptionBuildItem = this.findByName(propertyKey,
+                        configDescriptionBuildItems);
+
+                if (optionalConfigDescriptionBuildItem.isPresent()) {
+                    var item = optionalConfigDescriptionBuildItem.get();
+                    PermissionDescription permissionDescription = new PermissionDescription(
+                            propertyKey,
+                            configSource.getValue(propertyKey),
+                            new ConfigDescription(item.getPropertyName(),
+                                    formatJavadoc(cleanUpAsciiDocIfNecessary(item.getDocs())),
+                                    item.getDefaultValue(),
+                                    isSetByDevServices(devServicesLauncherConfig, item.getPropertyName()),
+                                    item.getValueTypeName(),
+                                    item.getAllowedValues(),
+                                    item.getConfigPhase().name()));
+                    configDescriptions.add(permissionDescription);
+                }
+            }
+        }
+
+        Map<AuthFieldType.AuthConfigType, List<PermissionSet>> result = configDescriptions.stream()
+                .collect(Collectors.groupingBy(
+                        PermissionDescription::getConfigType,
+                        Collectors.collectingAndThen(
+                                Collectors.groupingBy(PermissionDescription::getConfigId,
+                                        Collectors.mapping(d -> d, Collectors.toList())),
+                                configMap -> configMap.values().stream()
+                                        .map(PermissionSet::new)
+                                        .sorted()
+                                        .collect(Collectors.toList()))));
+
+        List<PermissionSet> permissionSets = result.get(AuthFieldType.AuthConfigType.PERMISSION);
+        List<PermissionSet> policyGroups = result.get(AuthFieldType.AuthConfigType.POLICY);
+
+        recorder.registerPermissionGroups(permissionSets, policyGroups);
+    }
+
+    @BuildStep(onlyIf = IsDevelopment.class)
+    @Record(ExecutionTime.RUNTIME_INIT)
+    void registerJsonRpcService(
+            List<ConfigDescriptionBuildItem> configDescriptionBuildItems,
+            BuildProducer<JsonRPCProvidersBuildItem> jsonRPCProvidersProducer,
+            BuildProducer<SyntheticBeanBuildItem> syntheticBeanProducer,
+            AuthDevUIRecorder recorder) {
+
+        DevConsoleManager.register("update-permission-set", map -> {
+            updateConfig(map);
+            register(AuthProcessor.configDescriptionBuildItems, AuthProcessor.devServicesLauncherConfig,
+                    AuthProcessor.recorder);
```

Comment:
you have recorder as a build step parameter and but you pass down the one stored on processor; these build steps can run in parallel. there can be race, please don't store the recorder on the processor